frontend/src/pages/notebookController/screens/server/SizeSelectField.tsx

```diff
@@ -0,0 +1,55 @@
+import * as React from 'react';
+import { FormGroup, Select, SelectOption } from '@patternfly/react-core';
+import AppContext from '../../../../app/AppContext';
+
+type SizeSelectFieldProps = {
+  value: string;
+  setValue: (newValue: string) => void;
+};
+
+const SizeSelectField: React.FC<SizeSelectFieldProps> = ({ value, setValue }) => {
+  const [sizeDropdownOpen, setSizeDropdownOpen] = React.useState<boolean>(false);
+  const { dashboardConfig } = React.useContext(AppContext);
+  console.log(value);
+
+  const sizeOptions = React.useMemo(() => {
+    const sizes = dashboardConfig?.spec?.notebookSizes;
+    if (!sizes?.length) {
+      return [<SelectOption key="Default" value="Default" description="No Size Limits" />];
+    }
+
+    return sizes.map((size) => {
+      const name = size.name;
+      const desc =
+        name === 'Default'
+          ? 'Resources set based on administrator configurations'
+          : `Limits: ${size?.resources?.limits?.cpu || '??'} CPU, ` +
+            `${size?.resources?.limits?.memory || '??'} Memory ` +
+            `Requests: ${size?.resources?.requests?.cpu || '??'} CPU, ` +
+            `${size?.resources?.requests?.memory || '??'} Memory`;
+      return <SelectOption key={name} value={name} description={desc} />;
+    });
+  }, [dashboardConfig]);
+
+  const handleSizeSelection = (e, selection) => {
```

Comment:
types please if you're not going to write this inline and let TS infer it from the PF component.

If you want to keep it separate, maybe grab the type from the Select Component:
```
type OnSelect = React.ComponentProps<typeof Select>['onSelect'];
```

You'll then have to handle the `string | SelectOptionObject` cases... and since you know that you are using a `string` value -- you can do `if (typeof selection === 'string') setValue(selection)`